File: Practica1/ej3/Receta.java
public class Receta {
    private String ingredientes;
    private String pasos;
    private int tiempoPreparacion;

    public Receta(String ingredientes, String pasos, int tiempoPreparacion) {
        this.ingredientes = ingredientes;
        this.pasos = pasos;
        this.tiempoPreparacion = tiempoPreparacion;
    }
    public Receta() {
        this.ingredientes = "";
        this.pasos = "";
        this.tiempoPreparacion = 0;
    }

    public void setIngredientes(String ingredientes) {
        this.ingredientes = ingredientes;
    }

    public void setPasos(String pasos) {
        this.pasos = pasos;
    }

    public void setTiempoPreparacion(int tiempoPreparacion) {
        this.tiempoPreparacion = tiempoPreparacion;
    }
    public String getIngredientes() {
        return ingredientes;
    }

    public String getPasos() {
        return pasos;
    }

    public int getTiempoPreparacion() {
        return tiempoPreparacion;
    }
}

File: practica2flutter/lib/receta.dart
import 'elemento.dart';

class Receta implements Elemento{

  late String ingredientes;
  late String pasos;
  late int tiempoPreparacion;

  @override
  String nombre = '';

  Receta(this.nombre) {
    ingredientes = '';
    pasos = '';
    tiempoPreparacion = 0;
  }
  void setIngredientes(String ingredientes) {
    this.ingredientes = ingredientes;
  }

  void setPasos(String pasos) {
    this.pasos = pasos;
  }

  void setTiempoPreparacion(int tiempoPreparacion) {
    this.tiempoPreparacion = tiempoPreparacion;
  }

  @override
  String toString() {
    return 'Ingredientes: $ingredientes\nPasos: $pasos\nTiempo de preparacion: $tiempoPreparacion minutos';
  }

  @override
  void add(Elemento elemento) {
    throw UnimplementedError();
  }
  @override
  void remove(Elemento elemento) {
    throw UnimplementedError();
  }
  @override
  String mostrar() {
    return toString();
  }

  @override
  String getNombre() {
    return nombre;
  }
  @override
  List<Elemento> getElementos() {
    throw UnimplementedError();
  }
}